references/organization/_06-Project-Plan-and-Workflow.md

```diff
@@ -0,0 +1,64 @@
+    - One based on a realistic causal graph with confounders. Preferably focus on a case with one latent confounder (like distance? Age? Gender?) <br>
+-	For each of the two causal graphs above, simulate choice data based on the outcome model assumed. <br>
+-	Estimate the choice model for each of those two datasets. <br>
+-	Repeat the process n times, and calculate the percentage of those times where the estimated parameters’ confidence intervals contain the true parameter. <br>
+
+### Dealing with latent confounders in the causal graph
+
+In this part of the work, we take the realistic causal graph from part 1 and remove a latent confounder from it. We then attempt to recover the latent confounder using the de-confounder algorithm proposed by Blei et al. (2018). We will apply this work to the simulated data set, where we know the true parameters of interest. We will repeat the same simulation process in part 1 where we sample from the population and estimate models where we control for the latent confounder using the deconfounder approach and look at the percent of time we recover the true parameter. 
+
+##### Concrete steps: 
+
+Functions <br>
+-	A class with functions to estimate factor models. Start with PPCA as in the deconfounder’s tutorial and expand as needed. <br>
+-	A function to perform posterior predictive checks on the factor model to assess its predictive accuracy. <br>
+
+Workflow <br>
+-	Start with a hypothesized general causal graph. For the simulated example, we’ll start by assuming the true causal graph used to generate the data, where we highlight the confounder as being latent/unobserved. <br>
+-	Then, come up with a “utility level” causal graph, where we come up with a causal graph for each utility equation since the confounding is at the level of the utilities in a MNL context. These causal graphs would inform the utility specifications, and will determine which utilities do and do not require estimating a latent confounder. We discussed this briefly before, but are we on the same page? <br>
+-	Based on the above specifications, estimate the required latent confounders using a factor model, start with PPCA. <br>
+-	Perform posterior predictive checks on the estimated factor model and assess goodness of fit. Since we’re doing this on a simulated dataset, then we know in advance how many latent confounders there are. We could potentially try controlling for more confounders and see how that affects the results (in other words, see how sensitive our analysis is to the number of confounders used). <br>
+-	Once the checks pass, include the latent confounder in the utility equations, and estimate a MNL. <br>
+-	Repeat the process n times, and calculate the percentage of those times where the estimated parameters’ confidence intervals contain the true parameter. <br>
+Finally, we can repeat the above process, but now on the full, real bike dataset. The workflow would be the same as above, with the addition of one step: <br>
```

Comment:
Addition of three steps actually. Once one has tried to falsify the causal graph and fit the model to the entire dataset (aka including the outcome model, i.e. the MNL with the latent confounders), one should check:
- the outcome model and
- the outcome + treatment-assignment-model (aka the model of the X's)

Luckily, both of these two are essentially just applying the techniques from my model checking paper and we can reuse the code from that project.

Reply:
While I agree on this, I don't necessarily 100% agree that it is necessary for the purposes of our work. Meaning -- if our models don't pass your checks, that conclusion wouldn't have a direct implication on the comparison of our approach vs. the no-causal graph/no latent confounder approach. I think this should be done for the eventual paper, but it's not directly relevant to the presentation.

Reply:
I'm with you, partially.

If we only talk about the simulated results, then I think you're right: the posterior predictive checks aren't **necessary** as we know we're simulating-from / estimating the data-generating-model.

If we speak about the real data though, I think the posterior predictive checks are necessary.

Assuming we find differences (as your prototype suggests we will) between the deconfounder based choice model, and the standard choice model, we still will not know whether or not we should put much stock in the deconfounder results.

In other words, our justification for drawing conclusions based on the deconfounder with real data is based on our model being a good fit to the real data, so I think we need to provide an abundance of evidence for the adequacy of the model fit (as with posterior predictive checks).